### PR TITLE
Add optional int list ODS type

### PR DESCRIPTION
**Pull request by gpetters94**

### CODE REVIEW

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 1]
```diff
@@ -226,6 +226,7 @@ def get_by_triple(self, key: Tuple[str, str, str]):
     "int": "Torch_IntType",
     "int[]": "TorchIntListType",
     "int?": "TorchOptionalIntType",
+    "int[]?": "TorchOptionalIntListType",
```
Comment (cathyzhyi) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py:229]: There is some confusion in naming. "Tensor?[]": "AnyTorchOptionalTensorListType" means the tensor element is optional but "int[]?": "TorchOptionalIntListType" means the list is optional.

Can you help to change all list type to ListOfXXXType. Optional list type becomes OptionalListOfXXXType and when list element is optional ListOfOptionalXXXType.

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

**Comment by gpetters94** [comment]
Are these names acceptable? Some of them are kind of unwieldy.

**Comment by cathyzhyi** [comment]
@silvasean Hey Sean, do you have any comments on the names? This is for addressing https://github.com/llvm/torch-mlir/pull/643#discussion_r819681492 to make the difference between `Tensor[]?` and `Tensor?[]`

**Comment by silvasean** [comment]
Following ODS conventions I think we should call these `AnyTorch*` (except the ones that are `Torch_X` naturally). So I would go with `Tensor[]?` as `AnyTorchOptionalListOfTensorType` and `Tensor?[]` as `AnyTorchListOfOptionalTensorType` (and `Tensor?[]?` would be `AnyTorchOptionalListOfOptionalTensorType`). I think the important thing is that they read left-to-right directly and consistently (so you just reverse the reading of the torch type string: `Tensor[]?` -> `Tensor, [], ?` --(reverse)--> `?, [], Tensor` -> `AnyTorchOptionalListOfTensorType`)

**Review by cathyzhyi — APPROVED** [review]

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 2]
```diff
@@ -20,15 +20,16 @@
...
     "Tensor?": "AnyTorchOptionalTensorType",
-    "Tensor?[]": "AnyTorchOptionalTensorListType",
-    "Tensor[]": "AnyTorchTensorListType",
+    "Tensor?[]": "AnyTorchListOfOptionalTensorType",
+    "Tensor[]": "AnyTorchListOfTensorType",
     "Scalar": "AnyTorchScalarType",
     "Scalar?": "AnyTorchOptionalScalarType",
     "int": "Torch_IntType",
-    "int[]": "TorchIntListType",
+    "int[]": "ListOfTorchIntType",
```
Comment (silvasean) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py:28]: AnyTorchListOfIntType

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 3]
```diff
@@ -20,15 +20,16 @@
...
     "int?": "TorchOptionalIntType",
```
Comment (silvasean) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py:29]: AnyTorchOptionalIntType

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py [hunk 4]
```diff
@@ -20,15 +20,16 @@
...
+    "int[]?": "OptionalListOfTorchIntType",
```
Comment (silvasean) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/torch_ods_gen.py:30]: AnyTorchOptionalListOfIntType

**Comment by cathyzhyi** [comment]
@gpetters94 the change looks good. You can go ahead and merge when you are ready.

**Pull request merged**